MotionMark/resources/statistics.js

```diff
@@ -223,6 +223,58 @@ Regression = Utilities.createClass(
         return this.s1 + this.t1 * complexity;
     },
 
+    _intersection: function(segment1, segment2)
+    {
+        return (segment1.s - segment2.s) / (segment2.t - segment1.t);
+    },
+
+    _error: function() {
+        return this.e1 + this.e2;
+    },
+
+    _areEssentiallyEqual: function(n1, n2) {
+        const epsilon = 0.0001;
+        return Math.abs(n1 - n2) < epsilon;
+    },
+
+    _setOptimal: function(segment1, segment2, x, xn, options) {
+        if (segment1.e + segment2.e > this.e1 + this.e2)
+            return false;
+
+        segment1.s = options.s1 !== undefined ? options.s1 : segment1.s;
+        segment1.t = options.t1 !== undefined ? options.t1 : segment1.t;
+        segment2.s = options.s2 !== undefined ? options.s2 : segment2.s;
+        segment2.t = options.t2 !== undefined ? options.t2 : segment2.t;
+
+        // The score is the x coordinate of the intersection of segment1 and segment2.
+        let complexity = this._intersection(segment1, segment2);
+
+        if (!this._areEssentiallyEqual(segment1.t, segment2.t)) {
+            // If segment1 and segment2 are not parallel, then they have to meet
+            // at complexity such that xp < complexity < x.
```

Comment:
The code now uses xn instead of xp.

Reply:
Yes this is right. x and xn represent two consecutive samples. So this comment should include the inequality ` x < complexity < xn`.